trpcultivate_phenotypes/src/TripalCultivateValidator/ValidatorTraits/FileTypes.php

```diff
@@ -0,0 +1,168 @@
+
+    // Extensions array must have a element.
+    if (empty($extensions)) {
+      throw new \Exception("The setSupportedMimeTypes() setter requires an array of file extensions that are supported by the importer and must not be empty.");
+    }
+
+    $mime_types = [];
+    $invalid_ext = [];
+
+    foreach($extensions as $ext) {
+      if (!isset(self::$extension_to_mime_mapping[ $ext ])) {
+        array_push($invalid_ext, $ext);
+        continue;
+      }
+      $mime_types = array_merge($mime_types, self::$extension_to_mime_mapping[$ext]);
+    }
+
+    if ($invalid_ext) {
+      $invalid_ext = implode(', ', $invalid_ext);
+      throw new \Exception('The setSupportedMimeTypes() setter does not recognize the following extensions: ' . $invalid_ext);
+    }
+
+    // Set the mime-types
+    $this->context['supported_mime_types'] = $mime_types;
+
+    // Set the supported file extensions
+    $this->context['file_extensions'] = $extensions;
+  }
+
+  /**
+   * Gets the supported file extensions of the current importer.
+   *
+   * @return array
+   *   The file extensions set by the setSupportedMimeTypes() setter method.
+   *
+   * @throws \Exception
+   *  - If the 'file_extensions' key does not exist in the context array
+   *    (ie. the setSupportedMimeTypes() method has NOT been called).
+   */
+  public function getSupportedFileExtensions() {
+
+    $context_key = 'file_extensions';
+
+    if (array_key_exists($context_key, $this->context)) {
+      return $this->context[ $context_key ];
+    }
+    else {
+      throw new \Exception('Cannot retrieve supported file extensions as they have not been set by setSupportedMimeTypes() method.');
+    }
+  }
+
+  /**
+   * Gets the supported file mime-types of the current importer.
+   *
+   * @return array
+   *   The file mime-types set by the setSupportedMimeTypes() setter method.
+   *
+   * @throws \Exception
+   *  - If the 'supported_mime_types' key does not exist in the context array
+   *    (ie. the setSupportedMimeTypes() method has NOT been called).
+   */
+  public function getSupportedMimeTypes() {
+
+    $context_key = 'supported_mime_types';
+
+    if (array_key_exists($context_key, $this->context)) {
+      return $this->context[ $context_key ];
+    }
+    else {
+      throw new \Exception('Cannot retrieve supported file mime-types as they have not been set by setSupportedMimeTypes() method.');
+    }
+  }
+
+  /**
+   * Gets the file mime-type of the input file.
+   *
+   * @return array
+   *   The file mime-type set by the setFileMimeType() setter method.
+   *
+   * @throws \Exception
+   *  - If the 'file_mime_type' key does not exist in the context array
+   *    (ie. the setFileMimeType() method has NOT been called).
+   */
+  public function getFileMimeType() {
+
+    $context_key = 'file_mime_type';
+
+    if (array_key_exists($context_key, $this->context)) {
+      return $this->context[$context_key];
+    } else {
```

Comment:
last one, If-else block same as the other.
```suggestion
    }
    else {
```

Reply:
Thank you for catching this! 👍